blackjack.el

```diff
@@ -676,46 +674,45 @@ Can be a single-character currency symbol such as \"$\", \"€\" or \"£\", or a
   "Return non-nil if DEALER-HAND up-card is an ace."
   (blackjack--is-ace-p (nth 1 (slot-value dealer-hand 'cards))))
 
+(defconst blackjack--card-value-hidden 13
+  "Token card \"value\" representing a hidden card.")
+
 (defun blackjack--draw-dealer-hand (game)
   "Draw the GAME dealer-hand."
-  (let* ((dealer-hand (slot-value game 'dealer-hand))
-         (cards (slot-value dealer-hand 'cards))
-         (hide-down-card (slot-value dealer-hand 'hide-down-card))
-         (card) (suit) (value))
+  (let ((dealer-hand (slot-value game 'dealer-hand)))
     (insert "  ")
-    (dotimes (x (length cards))
-      (setq card (nth x cards))
-      (if (and hide-down-card (= x 0))
-          (progn
-            (setq value 13)
-            (setq suit 0))
-        (setq value (slot-value card 'value))
-        (setq suit (slot-value card 'suit)))
-      (insert (blackjack--card-face game value suit))
-      (insert " "))
+    (seq-map-indexed
+     (lambda (card index)
+       "Display one of the cards in the dealer's hand."
+       (insert
+        (apply #'blackjack--card-face game
+               (if (and (= index 0) (slot-value dealer-hand 'hide-down-card))
+                   `(,blackjack--card-value-hidden 0)
+                 `(,(slot-value card 'value)
+                   ,(slot-value card 'suit)))))
+       (insert " "))
+     (slot-value dealer-hand 'cards))
     (insert " ⇒  ")
     (insert (number-to-string (blackjack--dealer-hand-value dealer-hand 'soft)))))
 
 (defun blackjack--dealer-hand-value (dealer-hand count-method)
   "Calculates DEALER-HAND cards total value based on COUNT-METHOD."
-  (let ((cards (slot-value dealer-hand 'cards))
-        (hide-down-card (slot-value dealer-hand 'hide-down-card))
-        (total 0) (card))
-    (dotimes (x (length cards))
-      (unless (and hide-down-card (= x 0))
-        (setq card (nth x cards))
-        (setq total (+ total (blackjack--card-val card count-method total)))))
+  (let ((total 0))
+    (seq-map-indexed
+     (lambda (card index)
+       "Value one of the cards in the dealer's hand."
+       (unless (and (= index 0) (slot-value dealer-hand 'hide-down-card))
+         (cl-incf total (blackjack--card-val card count-method total))))
+     (slot-value dealer-hand 'cards))
     (when (and (eq count-method 'soft) (> total 21))
       (setq total (blackjack--dealer-hand-value dealer-hand 'hard)))
     total))
 
 (defun blackjack--draw-player-hands (game)
   "Draw GAME players hands."
-  (let ((player-hands (slot-value game 'player-hands))
-        (player-hand))
-    (dotimes (x (length player-hands))
-      (setq player-hand (nth x player-hands))
-      (blackjack--draw-player-hand game player-hand x))))
+  (seq-map-indexed
+   (apply-partially #'blackjack--draw-player-hand game)
```

Comment:
How can this call to `blackjack--draw-player-hand` continue to work properly without the index value being passed?

Reply:
The index value is in fact being passed, "behind the scenes". Unlike other `*map*` functions, `seq-map-indexed` needs a function that takes *two* arguments, the second of which is the index. You don't have to declare or initialize the index, that's all taken care of for you by `seq-map-indexed`. `(info "(elisp) Sequence Functions")`.